command/variadic.go

```diff
@@ -0,0 +1,138 @@
+package command
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/peak/s5cmd/objurl"
+	"github.com/peak/s5cmd/storage"
+)
+
+// Arg is a container type for supporting variadic arguments.
+type Arg struct {
+	origSrc *objurl.ObjectURL
+	obj     *storage.Object
+}
+
+// expandSources returns the full list of objects from the given src arguments.
+// If src is an expandable URL, such as directory, prefix or a glob, all
+// objects are returned by walking the source. It expands multiple resources asynchronously
+// and returns read-only arg channel.
+func expandSources(
```

Comment:
This function well deserves a unit test.

Reply:
I will add both e2e and unit test

Reply:
I added multiple e2e tests for the use case.